### ISSUE TITLE
No Doco for this obsolete message 'Address' is obsolete: 'Use the string based overloads.'	

### ISSUE THREAD

**Issue by petersgiles** [Particular/docs.particular.net#1280]
I was upgrading NServiceBus.Distributor.MSMQ and got this message `Address' is obsolete: 'Use the string based overloads. Will be removed in version 7.0.0.'	`

I looked through the [upgrade guide ](http://docs.particular.net/nservicebus/upgrades/5to6 ) and wasn't able to find examples of what to do.

I looked at the NServiceBus v6 code no hints there

``` C#
    [ObsoleteEx(
        Message = "Use the string based overloads",
        RemoveInVersion = "7.0",
        TreatAsErrorFromVersion = "6.0")]
    public class Address
    {
    }
```

How do I `Use the string based overloads`?

Here is an example of it's use in NServiceBus.Distributor.MSMQ. It seems to be used to Parse and carry address info around. 

``` C#
 public static Address GetMasterNodeAddress(ReadOnlySettings settings)
        {
            var unicastBusConfig = settings.GetConfigSection<UnicastBusConfig>();

            //allow users to override data address in config
            if (!string.IsNullOrWhiteSpace(unicastBusConfig?.DistributorDataAddress))
            {
                return Address.Parse(unicastBusConfig.DistributorDataAddress);
            }

            var masterNode = GetMasterNode(settings);

            if (string.IsNullOrWhiteSpace(masterNode))
            {
                return Address.Parse(settings.EndpointName());
            }

            ValidateHostName(masterNode);

            return new Address(settings.EndpointName(), masterNode);
        }
```

Should there there be an example of what to do in the 5to6 upgrade guide? The code example here is using it to parse a config setting how is config setting parsing done now?

**Comment by DavidBoike** [comment]
@Particular/nservicebus

**Comment by DavidBoike** [comment]
Grr @Particular/nservicebus-maintainers

**Comment by timbussmann** [comment]
I think you just need to change `Address` o `string` here? What part of the posted code is used by the user?

**Comment by petersgiles** [comment]
@timbussmann I don't have access to customer code so I can't tell you anything other than this is part of our public API and therefore customers are likely to be using this. 

It seems to me that `Address.Parse(unicastBusConfig.DistributorDataAddress)` is a richer API than `string.Parse` otherwise why would we write it and `Address(settings.EndpointName(), masterNode)` is obviously doing something special.

I am working here utterly without context. Could you connect the PR or documentation where the decision was made to remove this from the API so I can catch up.

**Comment by DavidBoike** [comment]
Address.Parse existed to take somequeue@machine and split it on @. So the address class was a thin and useless wrapper around a string.

But this issue is about doco. The obsolete message should probably say "Wherever an Address instance was used previously, use a string in the form of QueueName or QueueName@ServerName.

And the upgrade guide should match.

The uses by the distributor are pretty specialized, to be fair.

**Comment by petersgiles** [comment]
Thank you @DavidBoike I can work with that. 

`public static Address GetMasterNodeAddress()` would become `public static string GetMasterNodeAddress()`

`return Address.Parse(unicastBusConfig.DistributorDataAddress);` would become 'return unicastBusConfig.DistributorDataAddress.Split('@')[0];'

**Comment by DavidBoike** [comment]
In second example, it wouldn't even be split.

**Comment by DavidBoike** [comment]
Unless it was Address.Parse(strAddress).QueueName or something

**Comment by SzymonPobiega** [comment]
Solved by #1371
